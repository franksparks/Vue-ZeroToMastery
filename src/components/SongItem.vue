<template>
  <div>
    <li
      class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
    >
      <div>
        <RouterLink
          :to="{ name: 'song', params: { id: song.docID } }"
          class="font-bold block text-gray-600"
          >{{ song.modifiedName }}</RouterLink
        >
        <span class="text-gray-500 text-sm">{{ song.genre }}</span>
      </div>

      <div class="text-gray-600 text-lg">
        <RouterLink
          custom
          :to="{ name: 'song', params: { id: song.docID }, hash: '#comments' }"
          v-slot="{ navigate }"
        >
          <span class="comments">
            {{ song.commentCount }}
            <span @click="navigate" v-icon.gray600="'comments'"></span>
          </span>
        </RouterLink>
      </div>
    </li>
  </div>
</template>

<script>
import { RouterLink } from "vue-router";

export default {
  name: "SongItem",
  props: ["song"],
  components: { RouterLink },
};
</script>
