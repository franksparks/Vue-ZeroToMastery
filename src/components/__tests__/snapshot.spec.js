import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import SongItem from "@/components/SongItem.vue";
import { expect } from "vitest";

describe("snapshot SongItem.vue", () => {
  test("renders correctly", () => {
    const song = {
      docID: "tabe",
      modifiedName: "test",
      displayName: "test",
      commentCount: 5,
    };

    const wrapper = shallowMount(SongItem, {
      props: { song },
      global: {
        components: {
          "router-link": RouterLinkStub,
        },
        directives: {
          icon: "icon",
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
