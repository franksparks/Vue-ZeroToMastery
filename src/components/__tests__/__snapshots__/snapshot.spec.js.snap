// Vitest Snapshot v1

exports[`snapshot SongItem.vue > renders correctly 1`] = `
<div>
  <li
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
    id="song-id-tabe"
  >
    <div>
      <router-link-stub
        ariacurrentvalue="page"
        class="font-bold block text-gray-600"
        custom="false"
        replace="false"
        to="[object Object]"
      />
      <span
        class="text-gray-500 text-sm"
      />
    </div>
    <div
      class="text-gray-600 text-lg"
    >
      <router-link-stub
        ariacurrentvalue="page"
        custom="true"
        replace="false"
        to="[object Object]"
      />
    </div>
  </li>
</div>
`;
